hook/genesis/govern.c

```diff
@@ -637,43 +637,48 @@ int64_t hook(uint32_t r)
             {
                 previous_member[0] = 'V';
 
-                for (int i = 1; GUARD(32), i < 32; ++i)
+                for (int tbl = 1; GUARD(2), tbl <= 2; ++tbl)
                 {
-                    previous_member[1] = i < 2 ? 'R' : i < 12 ? 'H' : 'S';
-                    previous_member[2] = 
-                        i == 0 ? 'R' :
-                        i == 1 ? 'D' :
-                        i < 12 ? i - 2 :
-                        i - 12;
-
-                    uint8_t vote_key[32];
-                    if (state(SBUF(vote_key), SBUF(previous_member)) == 32)
+                    for (int i = 0; GUARD(66), i < 32; ++i)
                     {
-                        uint8_t vote_count = 0;
-
-                        // find and decrement the vote counter
-                        vote_key[0] = 'C';
-                        vote_key[1] = previous_member[1];
-                        vote_key[2] = previous_member[2];
-                        if (state(&vote_count, 1, SBUF(vote_key)) == 1)
+                        previous_member[1] = i < 2 ? 'R' : i < 12 ? 'H' : 'S';
+                        previous_member[2] = 
+                            i == 0 ? 'R' :
+                            i == 1 ? 'D' :
+                            i < 12 ? i - 2 :
+                            i - 12;
+                        previous_member[3] = tbl;
+
+                        uint8_t vote_key[32];
+                        if (state(SBUF(vote_key), SBUF(previous_member)) > 0)
```

Comment:
This section seems to have to be my following code.
`if (state(vote_key + 12, 20, SBUF(previous_member)) == 20)`

if `if (state(SBUF(vote_key), SBUF(previous_member)) > 0)`
```
previous_member==:: 565310010000000000000000AE123A8556F3CF91154711376AFB0F894F832B3D
vote_key==:: 4353100156F3CF91154711376AFB0F894F832B3D000000000000000000000000
```

if `if (state(vote_key + 12, 20, SBUF(previous_member)) == 20)`,
```
previous_member==:: 565310010000000000000000AE123A8556F3CF91154711376AFB0F894F832B3D
vote_key==:: 435310010000000000000000AE123A8556F3CF91154711376AFB0F894F832B3D
```

---

It looks like I can leave `if (state(SBUF(vote_key), SBUF(previous_member)) > 0)` as is and just rewrite the first few bytes when vote_key matches previous_member's value, but I don't know why.

Reply:
`state(SBUF(vote_key), SBUF(previous_member))` returns 20 not 32

Reply:
but `sizeof(previous_member)` is 32

Reply:
It's the size of the data being written to the buffer. We will need to use special padding it looks like.

It can be 32 bytes for a hook, 8 bytes for an xfl or 20 bytes for an seat.